R/call_utils.R

```diff
@@ -49,108 +21,96 @@ call_check_parse_varname <- function(varname) {
-#' teal.slice:::call_condition_choice("SEX", choices = factor(c("F", "M")))
-#' teal.slice:::call_condition_choice("x$SEX", choices = Sys.Date())
-#' teal.slice:::call_condition_choice("trunc(x$SEX)", choices = Sys.time())
+#' \dontrun{
+#' call_condition_choice("SEX", choices = c(1, 2))
+#' call_condition_choice("SEX", choices = "F")
+#' call_condition_choice("SEX", choices = c("F", "M"))
+#' call_condition_choice("x$SEX", choices = Sys.Date())
+#' call_condition_choice("x$SEX", choices = Sys.time())
+#' }
 #' @return a `call`
 #' @keywords internal
 call_condition_choice <- function(varname, choices) {
-  varname <- call_check_parse_varname(varname)
-
-  if (is.factor(choices)) {
-    choices <- as.character(choices)
-  } else if (inherits(choices, "Date")) {
-    choices <- format(choices)
-  } else if (inherits(choices, c("POSIXct", "POSIXlt"))) {
-    choices <- format(choices)
-  }
-
+  checkmate::assert_string(varname)
+  checkmate::assert_multi_class(
+    choices,
+    c("integer", "numeric", "character", "factor", "Date", "POSIXct", "POSIXlt")
+  )
 
-  if (length(choices) == 1) {
-    call("==", varname, choices)
+  if (is.numeric(choices)) {
+    if (length(choices) == 1) {
+      sprintf("%s == %s", varname, sprintf("%s", choices))
+    } else {
+      sprintf("%s %%in%% c(%s)", varname, toString(sprintf("%s", choices)))
+    }
   } else {
-    c_call <- do.call(
-      "call",
-      append(list("c"), choices)
-    )
-    # c_call needed because it needs to be vector call
-    # instead of vector. SummarizedExperiment.subset
-    # handles only vector calls
-    call("%in%", varname, c_call)
+    if (length(choices) == 1) {
+      sprintf("%s == %s", varname, sprintf("\"%s\"", choices))
+    } else {
+      sprintf("%s %%in%% c(%s)", varname, toString(sprintf("\"%s\"", choices)))
+    }
   }
 }
 
 #' `numeric` range condition call
 #'
 #' Compose `numeric` range condition call from inputs
 #'
-#' @param varname (`name` or `character(1)`)\cr
+#' @param varname (`character(1)`)\cr
 #'   name of the variable
 #'
 #' @param range (`numeric(2)`)\cr
 #'   range of the variable
 #'
 #' @return call
 #' @examples
-#' teal.slice:::call_condition_range("AGE", range = c(1, 2))
-#' teal.slice:::call_condition_range(as.name("AGE"), range = c(-1.2, 2.1))
-#' teal.slice:::call_condition_range(
-#'   teal.slice:::call_extract_list("ADSL", "AGE"),
+#' \dontrun{
+#' call_condition_range("AGE", range = c(1, 2))
+#' call_condition_range(
+#'   call_extract_list("ADSL", "AGE"),
 #'   range = c(-1.2, 2.1)
 #' )
-#' @return a `call`
+#' }
+#' @return a character string
 #' @keywords internal
 call_condition_range <- function(varname, range) {
+  checkmate::assert_string(varname)
   checkmate::assert_numeric(range, len = 2, sorted = TRUE)
-  varname <- call_check_parse_varname(varname)
-  call(
-    "&",
-    call(">=", varname, range[1]),
-    call("<=", varname, range[2])
-  )
+
+  sprintf("%s >= %.4f & %s <= %.4f", varname, range[1], varname, range[2])
```

Comment:
You can use index to specify which argument to take: `%<index>$<format>` . But besides, I wonder if 4 digits is fine. This condition will be produced based on the slider selection - I believe `%.4f` rounds the values so it might cause some bugs if in shiny (reactive values) we will keep the better precision. Using calls was safer in this case because original number was a part of the call. I will think about this so we can discuss.

```suggestion
  sprintf("%1$s >= %2$.4f & %1$s <= %3$.4f", varname, range[1], range[2])
```